### PR TITLE
[Observability] Add claim relays counter

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Summary

- Add a new telemetry counter which tracks the number of relays as they move through the claim/proof lifecycle.
- Fixes broken claim/proof plots (broken in #629)

image
> NB: changes outlined in red.

## Issue

- #643 

## Type of change

Select one or more:

- [x] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

**Documentation changes** (only if making doc changes)
- [ ] `make docusaurus_start`; only needed if you make doc changes

**Local Testing** (only if making code changes)
- [ ] **Unit Tests**: `make go_develop_and_test`
- [ ] **LocalNet E2E Tests**: `make test_e2e`
- See [quickstart guide](https://dev.poktroll.com/developer_guide/quickstart) for instructions

**PR Testing** (only if making code changes)
- [ ] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.
    - **THIS IS VERY EXPENSIVE**, so only do it after all the reviews are complete.
    - Optionally run `make trigger_ci` if you want to re-trigger tests without any code changes
    - If tests fail, try re-running failed tests only using the GitHub UI as shown [here](https://github.com/pokt-network/poktroll/assets/1892194/607984e9-0615-4569-9452-4c730190c1d2)

## Sanity Checklist

- [ ] I have tested my changes using the available tooling
- [x] I have commented my code
- [x] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

## Summary by CodeRabbit

- **New Features**
  - Introduced separate telemetry counters for relays and compute units in claim tracking.

- **Refactor**
  - Refactored handling of settled and expired claims to use structured results, improving code clarity and maintainability.
  - Updated stress test scripts for consistent access to plans data structure.

- **Bug Fixes**
  - Standardized capitalization in Grafana dashboard expressions to ensure accurate metric filtering and aggregation.

- **Tests**
  - Enhanced load tests to use consistent references, improving reliability and consistency of test execution.

### CODE REVIEW

File: x/proof/keeper/msg_server_create_claim.go [hunk 1]
```diff
@@ -32,11 +32,8 @@ func (k msgServer) CreateClaim(
 		if !isExistingClaim {
 			// TODO_IMPROVE: We could track on-chain relays here with claim.GetNumRelays().
```
Comment (bryanchriswhite) [inline x/proof/keeper/msg_server_create_claim.go:33]:
```suggestion
```

File: telemetry/event_counters.go [hunk 2]
```diff
@@ -98,6 +98,34 @@ func ClaimComputeUnitsCounter(
...
+	incrementAmount := numRelays
+	labels := []metrics.Label{
+		{Name: "unit", Value: "relays"},
+		{Name: "claim_proof_stage", Value: claimProofStage.String()},
+	}
+
+	// Ensure the counter is not incremented if there was an error.
+	if err != nil {
+		incrementAmount = 0
+		labels = AppendErrLabel(err, labels...)
```
Comment (Olshansk) [inline telemetry/event_counters.go:119]: This is being removed elsewhere?

File: x/tokenomics/types/types.go [hunk 3]
```diff
@@ -1,3 +1,19 @@
 package types
 
 // This file is in place to declare the package for dynamically generated protobufs
+
+// PendingClaimsResult encapsulates the result of settling pending claims. It is
+// intended to be used to represent settled and expired results as unique instances.
+type PendingClaimsResult struct {
+	NumClaims           uint64
```
Comment (Olshansk) [inline x/tokenomics/types/types.go:8]: Thoughts on adding an iota on the result type that contained "Settled" or "Expired"?

File: x/tokenomics/types/types.go [hunk 4]
```diff
@@ -1,3 +1,19 @@
...
+	NumComputeUnits     uint64
+	NumRelays           uint64
+	RelaysPerServiceMap map[string]uint64
+}
+
+// NewClaimSettlementResult creates a new PendingClaimsResult.
+func NewClaimSettlementResult() PendingClaimsResult {
```
Comment (Olshansk) [inline x/tokenomics/types/types.go:15]: This would take in the iota introduced.

**Review by Olshansk — CHANGES_REQUESTED** [review]

File: telemetry/event_counters.go (hunk 2 above)
Comment (bryanchriswhite) [reply telemetry/event_counters.go:119]:
```suggestion
```

File: x/tokenomics/types/types.go (hunk 3 above)
Comment (bryanchriswhite) [reply x/tokenomics/types/types.go:8]: I'm not sure I follow. Can you elaborate?

File: x/tokenomics/types/types.go (hunk 4 above)
Comment (bryanchriswhite) [reply x/tokenomics/types/types.go:15]: :thinking:

**Review by red-0ne — APPROVED** [review]
LGTM

**Pull request merged**